File: first-duplicate-number/FirstDuplicateNumber.cs
using System;
using System.Collections.Generic;

public class FirstDuplicateNumber {
    public static int FirstDuplicateNum(int[] array) {
        Dictionary<int, int> dictionary = new Dictionary<int, int>();
        int index = 0;
        while(index < array.Length) {
            int current = dictionary.GetValueOrDefault(array[index]);
            if(current == 1) { return array[index]; }
            dictionary.Add(array[index], dictionary.GetValueOrDefault(array[index]) + 1);
            index++;
        }
        return -1;
    }

    public static int FirstDuplicateNum(int[] array) {
        HashSet<int> set = new HashSet<int>();
        foreach(var arr in array) {
            if(set.Contains(arr)) { return arr; }
            set.Add(arr);
        }
        return -1;
    }
}

File: first-duplicate-number/FirstDuplicateNumber.swift
import Foundation

class FirstDuplicateNumber {
    func firstDuplicateNumber1(_ numbers: [Int]) -> Int {
        var set: Set<Int> = []
        var index = 0
        while(index < numbers.count) {
            if(set.contains(numbers[index])) { return numbers[index] }
            set.insert(numbers[index])
            index += 1
        }
        return -1
    }

    func firstDuplicateNumber2(_ numbers: [Int]) -> Int {
        var dictionary: [Int: Int] = [:]
        for index in stride(from: 0, to: numbers.count, by: 1) {
            if let current = dictionary[numbers[index]] {
                return numbers[index]
            }
            dictionary[numbers[index]] = 0
        }
        return -1
    }
}